src/plugins/auto/src/plugin.cpp

```diff
@@ -910,39 +910,84 @@ std::vector<DeviceInformation> Plugin::filter_device(const std::vector<DeviceInf
 }
 
 std::vector<DeviceInformation> Plugin::filter_device_by_model(const std::vector<DeviceInformation>& meta_devices,
-                                                                const std::shared_ptr<const ov::Model>& model) const {
+                                                              const std::shared_ptr<const ov::Model>& model,
+                                                              PluginConfig& load_config) const {
     if (meta_devices.empty()) {
         OPENVINO_THROW("No available device to filter ", get_device_name(), " plugin");
     }
 
-    std::vector<DeviceInformation> filter_device;
-    auto is_stateful = [&]() {
-        for (auto& op : model->get_ops()) {
-            if (std::dynamic_pointer_cast<ngraph::op::AssignBase>(op) ||
-                std::dynamic_pointer_cast<ngraph::op::ReadValueBase>(op)) {
-                    LOG_INFO_TAG("stateful mode, try deployed to CPU");
-                    return true;
-                }
+    auto disable_startup_runtime_fallback = [&]() {
+        if (load_config.get_property(ov::intel_auto::enable_startup_fallback)) {
+            LOG_WARNING_TAG("Setting property ov::intel_auto::enable_startup_fallback to false for stateful model.");
+            load_config.set_property(ov::intel_auto::enable_startup_fallback(false));
+        }
+        if (load_config.get_property(ov::intel_auto::enable_runtime_fallback)) {
+            LOG_WARNING_TAG("Setting property ov::intel_auto::enable_running_fallback to false for stateful model.");
+            load_config.set_property(ov::intel_auto::enable_runtime_fallback(false));
         }
-        return false;
     };
 
+    if (meta_devices.size() == 1) {
+        disable_startup_runtime_fallback();
```

Comment:
if only one device, we don't need to disable this？

Reply:
Yes. Found the similar logic for one device.